content/en/docs/concepts/storage/volume-snapshots.md

```diff
@@ -35,38 +37,36 @@ Users need to be aware of the following when using this feature:
 
 * API Objects `VolumeSnapshot`, `VolumeSnapshotContent`, and `VolumeSnapshotClass` are CRDs, not part of the core API.
 * `VolumeSnapshot` support is only available for CSI drivers.
-* As part of the deployment process, the Kubernetes team provides a sidecar helper container for the snapshot controller called `external-snapshotter`. It watches `VolumeSnapshot` objects and triggers `CreateSnapshot` and `DeleteSnapshot` operations against a CSI endpoint.
+* As part of the deployment process in `VolumeSnapshot` API v1beta1, the Kubernetes team provides two external snapshot controllers called `external-snapshotter`, a common controller to be deployed on the controller node and a sidecar helper container to be deployed together with the CSI driver.  The common controller watches `VolumeSnapshot` and `VolumeSnapshotContent` objects and is responsible for the creation and deletion of `VolumeSnapshotContent` object in dynamic provisioning.  The sidecar controller watches `VolumeSnapshotContent` objects and triggers `CreateSnapshot` and `DeleteSnapshot` operations against a CSI endpoint.
```

Comment:
Optional suggestion (not particularly relevant to v1.17, but a good idea): add a glossary tooltip for “controller”:
```suggestion
* As part of the deployment process in `VolumeSnapshot` API v1beta1, the Kubernetes team provides two external snapshot {{< glossary_tooltip term_id="controller" text="controllesr" >}} called `external-snapshotter`, a common controller to be deployed into the control plane, and a sidecar helper container to be deployed together with the CSI driver.  The common controller watches `VolumeSnapshot` and `VolumeSnapshotContent` objects and is responsible for the creation and deletion of `VolumeSnapshotContent` object in dynamic provisioning.  The sidecar controller watches `VolumeSnapshotContent` objects and triggers `CreateSnapshot` and `DeleteSnapshot` operations against a CSI endpoint.
```

Reply:
I'm not sure I understand the “_in `VolumeSnapshot` API v1beta1_” addition. What's the learning outcome for a reader seeing that phrase?

Reply:
Aside: I've got an open PR (#16885) to add reference documentation for controllers. Would it make sense to put the implementation details into the reference section once that's merged?

Reply:
The snapshot controller code repo is here: https://github.com/kubernetes-csi/external-snapshotter, not in the core kubernetes repo.  Do you want to include out-of-tree controllers in the reference doc?

On the other hand, since we documented the snapshot controller here, it does make sense to reference it in your PR.

Reply:
I think it could be a separate PR and a section called something like “Additional controllers”, to highlight that this doesn't run inside `kube-controller-manager`.